eo-maven-plugin/src/test/java/org/eolang/maven/DiscoverMojoTest.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016-2022 Objectionary.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven;
+
+import com.yegor256.tojos.MnJson;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Deque;
+import java.util.LinkedList;
+import java.util.Map;
+import org.cactoos.Input;
+import org.cactoos.io.ResourceOf;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+/**
+ * Test case for {@link DiscoverMojo}.
+ *
+ * @since 0.28.11
+ */
+final class DiscoverMojoTest {
+
+    /**
+     * The name of foreign dependencies file.
+     */
+    private static final String EO_FOREIGN = "eo-foreign.json";
+
+    @ParameterizedTest
+    @CsvSource({
+        "org/eolang/maven/mess.eo, 7",
+        "org/eolang/maven/sum.eo, 0",
+        "org/eolang/maven/withwarning.eo, 1"
+    })
+    void executeDiscoveryPhaseForEoCorrectPrograms(
```

Comment:
@volodya-lombrozo just a small hint: it's better to name test methods as if you finish a sentence that start with "The object under test...". In your case it would be: "The object under test _execute_ discovery phase for eo correct programs". It's better to say "executes"

Reply:
@yegor256 Done. `executeDiscoveryPhaseForEoCorrectPrograms` -> `executesDiscoveryPhaseForCorrectEoPrograms`